group/outputs.tf

```diff
@@ -6,6 +6,34 @@ output "display_name" {
   value = azuread_group.group.display_name
 }
 
+output "group_with_roles" {
```

Comment:
I believe this adds a lot of overhead in terms of managing this output. I suggest just adding another output for role assignments:

```terraform
output "role_assignments" {
  value = azurerm_role_assignment.roles 
}
```